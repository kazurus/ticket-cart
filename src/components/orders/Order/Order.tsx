import React from 'react';

interface Props {
  id: number;
  createdAt: number;
  from: string;
  to: string;
  price: number;
}

const Order: React.FC<Props> = ({ id, createdAt, from, to, price }) => (
  <div>
    Order: {id} - {createdAt}
    <br />
    {from} - {to} &#36;{price}
  </div>
);

export default Order;
