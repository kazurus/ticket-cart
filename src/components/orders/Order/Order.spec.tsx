import React from 'react';
import { render } from '@testing-library/react';
import Order from './Order';

describe('Order component', () => {
  test('should render props', () => {
    const props = {
      id: 1,
      createdAt: 111,
      to: 'to',
      from: 'from',
      price: 10,
    };

    const { id, createdAt, to, from, price } = props;

    const { container } = render(
      <Order {...{ id, createdAt, to, from, price }} />
    );

    expect(container).toMatchSnapshot();
  });
});
