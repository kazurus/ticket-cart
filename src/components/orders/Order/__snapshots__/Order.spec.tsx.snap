// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order component should render props 1`] = `
<div>
  <div>
    Order: 
    1
     - 
    111
    <br />
    from
     - 
    to
     $
    10
  </div>
</div>
`;
